tests/test_macro.py

```diff
@@ -0,0 +1,153 @@
+def test_script_macro():
+    # Makes use of running script from absolute path.
+    engine, folder = create_engine()
+    with patch("autokey.model.Phrase.persist"), patch("autokey.model.Folder.persist"):
+        dummy_folder = autokey.model.Folder("dummy")
+        # This is a duplicate of the phrase added by the target script, but in a
+        # different folder.
+        dummy = engine.create_phrase(dummy_folder, "arg 1", "arg2", temporary=True)
+        assert_that(folder.items, not_(has_item(dummy)))
+
+        script = get_autokey_dir() + "/tests/create_single_phrase.py"
+        test = "<script name='{}' args='arg 1',arg2>".format(script)
+
+        expandMacro(engine, test)
+        assert_that(folder.items, has_item(dummy))
+
+def test_script_macro_spaced_quoted_args():
+    pass
+
+def test_cursor_macro():
+    engine, folder = create_engine()
+    test="one<cursor>two"
+    expected="onetwo<left><left><left>"
+    assert_that(expandMacro(engine, test), is_(equal_to(expected)),
+                "cursor macro returns wrong text")
+
+
+@unittest.mock.patch('datetime.datetime', FakeDate)
+def test_date_macro():
+    from datetime import datetime
+    FakeDate.now = classmethod(lambda cls: datetime(2019, 1, 1))
+
+    engine, folder = create_engine()
+    test="<date format=%d/%m/%y>"
+    expected="01/01/19"
+    assert_that(expandMacro(engine, test), is_(equal_to(expected)),
+                "Date macro fails to expand")
+
+def test_file_macro():
+    engine, folder = create_engine()
+    path =  get_autokey_dir() + "/tests/dummy_file.txt"
+    test="<file name={}>".format(path)
+    expected="test result macro expansion\n"
+    assert_that(expandMacro(engine, test), is_(equal_to(expected)),
+                "file macro does not expand correctly")
+
+def test_macro_expansion():
+    engine, folder = create_engine()
+    path =  get_autokey_dir() + "/tests/dummy_file.txt"
+    contents="middle <file name={}> macro".format(path, path)
+    expected="middle test result macro expansion\n macro".format(path, path)
+    assert_that(expandMacro(engine, contents), is_(equal_to(expected)),
+                "Macros between other parts don't expand properly")
+    contents="<file name={}> two macros this time <file name={}>".format(path, path)
+    expected="test result macro expansion\n two macros this time test result macro expansion\n".format(path, path)
+    assert_that(expandMacro(engine, contents), is_(equal_to(expected)),
+                "Two macros per phrase don't expand properly")
+    contents="<file name={}> mixed macro <cursor> types".format(path)
+    expected="test result macro expansion\n mixed macro  types<left><left><left><left><left><left>"
+    assert_that(expandMacro(engine, contents), is_(equal_to(expected)),
+                "mixed macros don't expand properly")
+
+# def test_nested_macro_raises_error():
+#     contents="<date format=<cursor>>"
```

Comment:
That may be a valid use case, though. Either having a literal `<cursor>` in your formatted date or expecting the cursor to be placed at the indicated position inside the formatted string.
Either way, this can be an error for now.